py/server/deephaven/liveness_scope.py

```diff
@@ -27,44 +50,143 @@ def __init__(self, j_scope: jpy.JType):
         self.j_scope = j_scope
 
     def __enter__(self):
+        warn('Instead of passing liveness_scope() to with, call open() on it first',
+             DeprecationWarning, stacklevel=2)
+        _push(self.j_scope)
         return self
 
     def __exit__(self, exc_type, exc_val, exc_tb):
-        self.close()
+        _pop(self.j_scope)
+        self.release()
+
+    def release(self):
+        """Closes the LivenessScope and releases all the query graph resources.
+
+        Raises:
+            DHError
+        """
+        try:
+            self.j_scope.release()
+            self.j_scope = None
+        except Exception as e:
+            raise DHError(e, message="failed to close the LivenessScope.")
 
     def close(self):
         """Closes the LivenessScope and releases all the query graph resources.
 
         Raises:
             DHError
         """
-        if self.j_scope:
-            try:
-                _JLivenessScopeStack.pop(self.j_scope)
-                self.j_scope.release()
-                self.j_scope = None
-            except Exception as e:
-                raise DHError(e, message="failed to close the LivenessScope.")
+        warn('This function is deprecated, prefer release(). Use cases that rely on this are likely to now fail.',
+             DeprecationWarning, stacklevel=2)
+        _pop(self.j_scope)
+        self.release()
+
+    @contextlib.contextmanager
+    def open(self, release_after_block: bool = True) -> "LivenessScope":
```

Comment:
Both "work" (after `s/LivenessScope/livenessScope/`), in that the first one will warn "you're doing it wrong, this will not work correctly", thus "strongly encouraging" the user to call open(). We can outright remove this right away and do away with deprecation warnings, or wait a version or two for anyone who happens to be using this and needs to transition away.

The new version is strictly more powerful, as it doesn't conflate creating/releasing a scope with adding/removing it on the stack. We can't really have it both ways... The other option (discussed with Ryan) is to make open() an optional step so that you _can_ call it either way, but this makes the Java and Python usage different/confusing going back and forth.

One other option is that we deprecate liveness_scope() itself, and make the LivenessScope() constructor usable, so we can leave liveness_scope() intact.